src/main/webapp/script.js

```diff
@@ -10,4 +10,75 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(getChart);
+
+const CHART_WIDTH = 800;
+const CHART_HEIGHT = 400;
+
+function getYouTubeComments() { 
+  const urlInput = document.getElementById('url-entry');
+  var url = cleanseUrl(urlInput.value);
+  fetch("/YouTubeComments?url="+url)
+      .then(response => response.json()).then((comments) =>{
+        return comments;
+    });
+}
+
+/*
+ * Extracts video id from full url
+ */ 
+function cleanseUrl(url) {
+  // Split web address from parameters, extract first parameter
+  var videoId = url.split("?");
+  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
+
+  // Remove parameter name to isolate video Id.
+  videoId = videoId.replace("v=", "");
+  
```

Comment:
extra line here?

Reply:
I feel like having the extra line spaces out the two steps of processing the url input from the return statement. If someone disagrees, though, I'll remove it.